Mathlib/Analysis/SpecialFunctions/Pow/NNReal.lean

```diff
@@ -70,6 +84,43 @@ theorem rpow_add {x : ℝ≥0} (hx : x ≠ 0) (y z : ℝ) : x ^ (y + z) = x ^ y
 theorem rpow_add' (x : ℝ≥0) {y z : ℝ} (h : y + z ≠ 0) : x ^ (y + z) = x ^ y * x ^ z :=
   NNReal.eq <| Real.rpow_add' x.2 h
 
+lemma rpow_add_intCast (hx : x ≠ 0) (y : ℝ) (n : ℤ) : x ^ (y + n) = x ^ y * x ^ n := by
+  ext; exact Real.rpow_add_intCast (mod_cast hx) _ _
+
+lemma rpow_add_natCast (hx : x ≠ 0) (y : ℝ) (n : ℕ) : x ^ (y + n) = x ^ y * x ^ n := by
+  ext; exact Real.rpow_add_natCast (mod_cast hx) _ _
+
+lemma rpow_sub_intCast (hx : x ≠ 0) (y : ℝ) (n : ℕ) : x ^ (y - n) = x ^ y / x ^ n := by
+  ext; exact Real.rpow_sub_intCast (mod_cast hx) _ _
+
+lemma rpow_sub_natCast (hx : x ≠ 0) (y : ℝ) (n : ℕ) : x ^ (y - n) = x ^ y / x ^ n := by
+  ext; exact Real.rpow_sub_natCast (mod_cast hx) _ _
+
+lemma rpow_add_intCast' {n : ℤ} (h : y + n ≠ 0) (x : ℝ≥0) : x ^ (y + n) = x ^ y * x ^ n := by
+  ext; exact Real.rpow_add_intCast' (mod_cast x.2) h
+
+lemma rpow_add_natCast' {n : ℕ} (h : y + n ≠ 0) (x : ℝ≥0) : x ^ (y + n) = x ^ y * x ^ n := by
+  ext; exact Real.rpow_add_natCast' (mod_cast x.2) h
+
+lemma rpow_sub_intCast' {n : ℤ} (h : y - n ≠ 0) (x : ℝ≥0) : x ^ (y - n) = x ^ y / x ^ n := by
+  ext; exact Real.rpow_sub_intCast' (mod_cast x.2) h
+
+lemma rpow_sub_natCast' {n : ℕ} (h : y - n ≠ 0) (x : ℝ≥0) : x ^ (y - n) = x ^ y / x ^ n := by
+  ext; exact Real.rpow_sub_natCast' (mod_cast x.2) h
+
+lemma rpow_add_one (hx : x ≠ 0) (y : ℝ) : x ^ (y + 1) = x ^ y * x := by
+  simpa using rpow_add_natCast hx y 1
+
+lemma rpow_sub_one (hx : x ≠ 0) (y : ℝ) : x ^ (y - 1) = x ^ y / x := by
+  simpa using rpow_sub_natCast hx y 1
+
+-- TODO: Swap argument order to `rpow_add'`/`rpow_sub'`
```

Comment:
This looks like it's suggesting changing the argument order here to match `rpow_add'` and `rpow_sub'`. Is that what you mean, or do you want to change the argument order _of_ `rpow_add`' and `rpow_sub'`? 
If you do want to change the argument order here, why not just do it here? :)

Reply:
It seems you misunderstood my comment: The specific scenario I was excluding was exactly the change you made. Can you justify why you think the argument order should be changed here?

Reply:
The data `x : NNReal` was coming before the prop `y + z ≠ 0`, which is annoying for rewriting purposes as you always need to write one underscore before being able to provide the proof obligation